static/js/protein/data_processing_utils.ts

```diff
@@ -0,0 +1,448 @@
+import _ from "lodash";
+
+import { GraphNodes } from "../shared/types";
+import { ProteinStrData } from "./chart";
+import { ProteinNumData } from "./chart";
+import { ProteinVarType } from "./page";
+import { InteractingProteinType } from "./page";
+
+const VARIANT_CATEGORY = [
+  "GeneticVariantFunctionalCategoryUTR3",
+  "GeneticVariantFunctionalCategoryUTR5",
+  "GeneticVariantFunctionalCategoryCodingSynon",
+  "GeneticVariantFunctionalCategoryFrameshift",
+  "GeneticVariantFunctionalCategoryIntron",
+  "GeneticVariantFunctionalCategoryMissense",
+  "GeneticVariantFunctionalCategoryNearGene3",
+  "GeneticVariantFunctionalCategoryNearGene5",
+  "GeneticVariantFunctionalCategoryNonsense",
+  "GeneticVariantFunctionalCategorySplice3",
+  "GeneticVariantFunctionalCategorySplice5",
+  "GeneticVariantFunctionalCategoryStopLoss",
+  "GeneticVariantFunctionalCategoryUnknown",
+  "GeneticVariantFunctionalCDSIndel",
+  "GeneticVariantFunctionalCategoryCDSReference",
+  "GeneticVariantFunctionalCategoryncRNA",
+];
+
+const VARIANT_AFFECTS = [
+  "ClinSigAffects",
+  "ClinSigAssociation",
+  "ClinSigAssociationNotFound",
+  "ClinSigBenign",
+  "ClinSigBenignLikelyBenign",
+  "ClinSigConflictingPathogenicity",
+  "ClinSigDrugResponse",
+  "ClinSigHistocompatability",
+  "ClinSigLikelyBenign",
+  "ClinSigLikelyPathogenic",
+  "ClinSigNotProvided",
+  "ClinSigOther",
+  "ClinSigPathogenic",
+  "ClinSigPathogenicLikelyPathogenic",
+  "ClinSigProtective",
+  "ClinSigRiskFactor",
+  "ClinSigUncertain",
+  "ClinSigUnknown",
+  "ClinSigUntested",
+];
+
+const CHEM_RELATIONS = [
+  "RelationshipAssociationTypeAssociated",
+  "RelationshipAssociationTypeNotAssociated",
+  "RelationshipAssociationTypeAmbiguous",
+];
+
+export function getTissueScore(data: GraphNodes): ProteinStrData[] {
+  // Tissue to score mapping.
+  if (!data) {
+    return [];
+  }
+  const returnData: ProteinStrData[] = [];
+  // check for null values
+  if (_.isEmpty(data.nodes) || _.isEmpty(data.nodes[0].neighbors)) {
+    return [];
+  }
+  for (const neighbour of data.nodes[0].neighbors) {
+    if (neighbour.property !== "detectedProtein") {
+      continue;
+    }
+    for (const node of neighbour.nodes) {
+      let tissue = null;
+      let score = null;
+      // check for null or non-existent property values
+      if (_.isEmpty(node.neighbors)) {
+        continue;
+      }
+      for (const n of node.neighbors) {
+        if (n.property === "humanTissue") {
+          // check for null tissue value and move to next iteration if true
+          if (_.isEmpty(n.nodes[0].value)) {
```

Comment:
you should also be checking that n.nodes is not empty before accessing the first element

whenever you are accessing an element in a list like this, you should check that the list has at least that many elements otherwise you might get an error for accessing an index not in the array.

Reply:
Great catch ! Done !

Reply:
nit: you can put the two checks together like: 
`if (_.isEmpty(n.nodes) || _.isEmpty(n.nodes[0].value)) {continue;}`

And everywhere else you made this change